plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineProjectWizard.java

```diff
@@ -38,17 +38,18 @@
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.ide.undo.WorkspaceUndoUtil;
 
-public abstract class AppEngineProjectWizard extends Wizard implements INewWizard {
+public abstract class AppEngineProjectWizard<T extends AppEngineWizardPage> extends Wizard
```

Comment:
It makes sense for a StandardWizard to have a StandardWizardPage and a FlexWizard to have a FlexWizardPage.  The alternative to templating the wizard page class is either pushing all the data fields into the base class, or having to cast (which is little different to templating except that it loses compiler type-checking).